### PR TITLE
[5.0] Use stringy functions

### PR DESCRIPTION
**Pull request by lucasmichot**

### CODE REVIEW

**Comment by Anahkiasen** [comment]
Can't we just make Str extend StaticStringy?

**Comment by lucasmichot** [comment]
@Anahkiasen IMO, this would produce a lot of new str:: functions and may lead to confusion

**Comment by GrahamCampbell** [comment]
@Anahkiasen I suggested this already.

File: src/Illuminate/Support/Str.php [hunk 1]
```diff
@@ -65,12 +65,7 @@ public static function camel($value)
 	 */
 	public static function contains($haystack, $needles)
 	{
-		foreach ((array) $needles as $needle)
-		{
-			if ($needle != '' && strpos($haystack, $needle) !== false) return true;
-		}
-
-		return false;
+		return StaticStringy::containsAny($haystack, array_filter((array) $needles));
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Str.php]: `array_filter((array) $needles)` is that not overkill?

Comment (GrahamCampbell) [inline src/Illuminate/Support/Str.php]: In most cases we'd get better performance with out this

File: src/Illuminate/Support/Str.php [hunk 2]
```diff
@@ -133,7 +128,7 @@ public static function is($pattern, $value)
 	 */
 	public static function length($value)
 	{
-		return mb_strlen($value);
+		return StaticStringy::length($value);
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Str.php]: mb_strlen is fine

Comment (GrahamCampbell) [inline src/Illuminate/Support/Str.php]: no need to add an extra call

File: src/Illuminate/Support/Str.php [hunk 3]
```diff
@@ -159,7 +154,7 @@ public static function limit($value, $limit = 100, $end = '...')
 	 */
 	public static function lower($value)
 	{
-		return mb_strtolower($value);
+		return StaticStringy::toLowerCase($value);
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Str.php]: samehere

**Comment by GrahamCampbell** [comment]
This really isn't required. Can you send only the contains and starts with changes without the array_filter stuff?

**Comment by lucasmichot** [comment]
Array_filter is necessary as the test would not pass with null or empty
needle

Le vendredi 2 janvier 2015, Graham Campbell <notifications@github.com> a
écrit :

**Comment by GrahamCampbell** [comment]
Ok, then, just don't send the pointless changes that nest calls to the same function.

**Pull request closed without merging**